collector/container/konflux-slim.Dockerfile

```diff
@@ -96,15 +92,37 @@ RUN ./builder/install/install-dependencies.sh && \
     ctest -V --test-dir ${CMAKE_BUILD_DIR} && \
     strip -v --strip-unneeded "${CMAKE_BUILD_DIR}/collector/collector"
 
-FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+# Application
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest AS ubi-minimal
+# The installer must be ubi (not minimal) and must be 8.9 or later since the earlier versions complain:
+#  subscription-manager is disabled when running inside a container. Please refer to your host system for subscription management.
+FROM ubi-normal AS rpm-implanter-app
```

Comment:
You could use the same `rpm-implanter` for both builder and app, but I would not insist, and besides we have a blocker problem to solve now with EC checks.

Reply:
If I understand your suggestion correctly and installed all dependencies in a single `rpm-implanter` stage, we would have all build time dependencies (like cmake, git, gcc, ...) in the final image. 
That would allow someone who has access to the container to compile new content, and cause more pain with CVEs. Additionally, the final image would be bigger.
Therefore, I would keep them as separate stages.

Reply:
That said, I merged the `rpm-installer-builder` and `builder` stages into one (https://github.com/stackrox/collector/pull/1573/commits/82f38460032ababceb355fe64d60115dae206574).

Reply:
I meant this
https://github.com/stackrox/stackrox/blob/25beb9ce8cf6da5bdd88fd5e20bc960c2ce7426e/scripts/konflux/subscription-manager/bro.self-test-multiple-targets.Dockerfile#L13-L34

Reply:
Thanks for clarifying, I hadn't thought about that. I will keep it as is though.